lms/models/application_instance.py

```diff
@@ -16,47 +18,139 @@
-        JSONSetting("jstor", "enabled", asbool),
-        JSONSetting("jstor", "site_code"),
-        JSONSetting("youtube", "enabled", asbool),
-        JSONSetting("hypothesis", "notes", name="Notes"),
-        JSONSetting(
-            "hypothesis",
-            "auto_assigned_to_org",
-            asbool,
-            name="Auto Assigned To Organisation",
-        ),
-        JSONSetting("hypothesis", "instructor_email_digests_enabled", asbool),
-        JSONSetting("hypothesis", "lti_13_sourcedid_for_grading", asbool),
-    )
+    class Settings(StrEnum):
+        BLACKBOARD_FILES_ENABLED = "blackboard.files_enabled"
+        BLACKBOARD_GROUPS_ENABLED = "blackboard.groups_enabled"
+
+        CANVAS_SECTIONS_ENABLED = "canvas.sections_enabled"
+        CANVAS_GROUPS_ENABLED = "canvas.groups_enabled"
+        CANVAS_FILES_ENABLED = "canvas.files_enabled"
+        CANVAS_FOLDERS_ENABLED = "canvas.folders_enabled"
+        CANVAS_STRICT_SECTION_MEMBERSHIP = "canvas.strict_section_membership"
+        CANVAS_PAGES_ENABLED = "canvas.pages_enabled"
+
+        CANVAS_STUDIO_ADMIN_EMAIL = "canvas_studio.admin_email"
+        CANVAS_STUDIO_CLIENT_ID = "canvas_studio.client_id"
+        CANVAS_STUDIO_CLIENT_SECRET = "canvas_studio.client_secret"  # noqa: S105
+        CANVAS_STUDIO_DOMAIN = "canvas_studio.domain"
+
+        D2L_CLIENT_ID = "desire2learn.client_id"
+        D2L_CLIENT_SECRET = "desire2learn.client_secret"  # noqa: S105
+        D2L_GROUPS_ENABLED = "desire2learn.groups_enabled"
+        D2L_FILES_ENABLED = "desire2learn.files_enabled"
+
+        GOOGLE_DRIVE_FILES_ENABLED = "google_drive.files_enabled"
+        MICROSOFT_ONEDRIVE_FILES_ENABLED = "microsoft_onedrive.files_enabled"
+
+        MOODLE_API_TOKEN = "moodle.api_token"  # noqa: S105
+        MOODLE_GROUPS_ENABLED = "moodle.groups_enabled"
+        MOODLE_FILES_ENABLED = "moodle.files_enabled"
+        MOODLE_PAGES_ENABLED = "moodle.pages_enabled"
+
+        VITALSOURCE_ENABLED = "vitalsource.enabled"
+        VITALSOURCE_USER_LTI_PARAM = "vitalsource.user_lti_param"
+        VITALSOURCE_USER_LTI_PATTERN = "vitalsource.user_lti_pattern"
+        VITALSOURCE_API_KEY = "vitalsource.api_key"
+        VITALSOURCE_STUDENT_PAY_ENABLED = "vitalsource.student_pay_enabled"
+
+        JSTOR_ENABLED = "jstor.enabled"
+        JSTOR_SITE_CODE = "jstor.site_code"
+
+        YOUTUBE_ENABLED = "youtube.enabled"
+
+        HYPOTHESIS_NOTES = "hypothesis.notes"
+        HYPOTHESIS_AUTO_ASSIGNED_TO_ORG = "hypothesis.auto_assigned_to_org"
+        HYPOTHESIS_INSTRUCTOR_EMAIL_DIGESTS_ENABLED = (
+            "hypothesis.instructor_email_digests_enabled"
+        )
+        HYPOTHESIS_LTI_13_SOURCEDID_FOR_GRADING = (
+            "hypothesis.lti_13_sourcedid_for_grading"
+        )
+
+    fields: Mapping[Settings, JSONSetting] = {
+        Settings.BLACKBOARD_FILES_ENABLED: JSONSetting(
+            Settings.BLACKBOARD_FILES_ENABLED, asbool
```

Comment:
The key of `fields` always matches the first argument to the `JSONSetting` constructor, which is the path of the setting in the DB's JSON blob. That's consistent, but in principle they could be different, right?

Tangential to what is happening in this PR, but the documentation for `JSONSetting` isn't clear about how `format` is used and what the legal values are. It seems that code in `search.py` expects it to be a function that converts a string representation of the value to the JSON type stored in the DB. It would make more sense to me if the `format` value were an enum. We only support a small number of setting types (strings, secrets, booleans).

Reply:
> The key of fields always matches the first argument to the JSONSetting constructor, which is the path of the setting in the DB's JSON blob. That's consistent, but in principle they could be different, right?

Yeah, I already broke the rule for D2L_XX (vs desire2learn_X)

Reply:
> It would make more sense to me if the format value were an enum.
Absolutely, made a change in that direction.